src/NetAppFiles/NetAppFiles/Pool/GetNetAppFilesPool.cs

```diff
@@ -0,0 +1,103 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.NetAppFiles.Common;
+using Microsoft.Azure.Commands.NetAppFiles.Helpers;
+using Microsoft.Azure.Commands.NetAppFiles.Models;
+using Microsoft.Azure.Management.NetApp;
+using System.Linq;
+
+namespace Microsoft.Azure.Commands.NetAppFiles.Pool
+{
+    [Cmdlet(
+        "Get",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesPool",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesPool))]
+    [Alias("Get-AnfPool")]
+    public class GetAzureRmNetAppFilesPool : AzureNetAppFilesCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group of the ANF pool",
+            ParameterSetName = FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the ANF account",
+            ParameterSetName = FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter("Microsoft.NetApp/netAppAccounts", nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the ANF pool",
+            ParameterSetName = FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias("PoolName")]
+        [ResourceNameCompleter("Microsoft.NetApp/netAppAccounts/capacityPools", nameof(ResourceGroupName), nameof(AccountName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource id of the ANF pool",
+            ParameterSetName = ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The account object containing the pool to return")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesAccount AccountInputObject { get; set; }
```

Comment:
@leonardbf nit: I would rename this to just `AccountObject`

Reply:
Ok. Renamed.